core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AllowanceValue.java

```diff
@@ -1,7 +1,15 @@
 package fr.sncf.osrd.envelope_sim.allowances;
 
 public abstract class AllowanceValue {
-    public abstract double getAllowanceTime(double baseTime, double distance, double totalDistance);
+    /** Returns the allowance, given the total time and distance of the trip */
+    public abstract double getAllowanceTime(double baseTime, double distance);
+
+    /** Returns the allowance for a part of a trip, divided over time */
+    public final double getPartialAllowanceTime(double sectionTime, double totalTime, double totalDistance) {
+        var totalAllowance = getAllowanceTime(totalTime, totalDistance);
+        var ratio = sectionTime / totalTime;
+        return totalAllowance * ratio;
+    }
```

Comment:
In the case of a TimeperDistance allowanceValue, this method will not return a margin time between stops which is proportional to the distance between stops